src/hooks/useSearchQuery.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+export const useSearchQuery = () => {
+  const [query, setQuery] = useState('');
```

Comment:
There is unnecessary complexity in having a local state for the search param as well. You don't need it actually and you can just rely on the value from the search param